bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/enums/HeatpumpVisibility.java

```diff
@@ -0,0 +1,378 @@
+    SYSEIN_VD2_ZEIT_MIN(258, "SysEin_VD2_Zeit_Min"),
+    EINSTTEMP_HYSTERESE_HR_VERKUERZT(259, "EinstTemp_Hysterese_HR_verkuerzt"),
+    EINST_LUF_FEUCHTESCHUTZ_AKT(260, "Einst_Luf_Feuchteschutz_akt"),
+    EINST_LUF_REDUZIERT_AKT(261, "Einst_Luf_Reduziert_akt"),
+    EINST_LUF_NENNLUEFTUNG_AKT(262, "Einst_Luf_Nennlueftung_akt"),
+    EINST_LUF_INTENSIVLUEFTUNG_AKT(263, "Einst_Luf_Intensivlueftung_akt"),
+    TEMPERATUR_LUEFTUNG_ZULUFT(264, "Temperatur_Lueftung_Zuluft"),
+    TEMPERATUR_LUEFTUNG_ABLUFT(265, "Temperatur_Lueftung_Abluft"),
+    OUT_ANALOG_3(266, "OUT_Analog_3"),
+    OUT_ANALOG_4(267, "OUT_Analog_4"),
+    IN_ANALOG_2(268, "IN_Analog_2"),
+    IN_ANALOG_3(269, "IN_Analog_3"),
+    IN_SAX(270, "IN_SAX"),
+    OUT_VZU(271, "OUT_VZU"),
+    OUT_VAB(272, "OUT_VAB"),
+    OUT_VSK(273, "OUT_VSK"),
+    OUT_FRH(274, "OUT_FRH"),
+    KUHLTEMP_SOLLTEMPMK3(275, "KuhlTemp_SolltempMK3"),
+    KUHLTEMP_ATDIFFMK3(276, "KuhlTemp_ATDiffMK3"),
+    IN_SPL(277, "IN_SPL"),
+    SYSEIN_LUEFTUNGSSTUFEN(278, "SysEin_Lueftungsstufen"),
+    SYSEIN_MELDUNG_TDI(279, "SysEin_Meldung_TDI"),
+    SYSEIN_TYP_WZW(280, "SysEin_Typ_WZW"),
+    BACNET(281, "BACnet"),
+    SPRACHEN_SLOWAKISCH(282, "Sprachen_SLOWAKISCH"),
+    SPRACHEN_LETTISCH(283, "Sprachen_LETTISCH"),
+    SPRACHEN_FINNISCH(284, "Sprachen_FINNISCH"),
+    KALIBRIERUNG_LWD(285, "Kalibrierung_LWD"),
+    IN_DURCHFLUSS(286, "IN_Durchfluss"),
+    LIN_ANSAUG_VERDICHTER(287, "LIN_ANSAUG_VERDICHTER"),
+    LIN_VDH(288, "LIN_VDH"),
+    LIN_UH(289, "LIN_UH"),
+    LIN_DRUCK(290, "LIN_Druck"),
+    EINST_SOLLWERT_TRL_KUEHLEN(291, "Einst_Sollwert_TRL_Kuehlen"),
+    ENTL_EXVENTIL(292, "Entl_ExVentil"),
+    EINST_MEDIUM_WAERMEQUELLE(293, "Einst_Medium_Waermequelle"),
+    EINST_MULTISPEICHER(294, "Einst_Multispeicher"),
+    EINST_MINIMALE_RUECKLAUFSOLLTEMPERATUR(295, "Einst_Minimale_Ruecklaufsolltemperatur"),
+    EINST_PKUEHLTIME(296, "Einst_PKuehlTime"),
+    SPRACHEN_TUERKISCH(297, "Sprachen_TUERKISCH"),
+    RBE(298, "RBE"),
+    EINST_LUF_STUFEN_FAKTOR(299, "Einst_Luf_Stufen_Faktor"),
+    FREIGABE_ZEIT_ZWE(300, "Freigabe_Zeit_ZWE"),
+    EINST_MIN_VL_KUEHL(301, "Einst_min_VL_Kuehl"),
+    ZWE1(302, "ZWE1"),
+    ZWE2(303, "ZWE2"),
+    ZWE3(304, "ZWE3"),
+    SEC(305, "SEC"),
+    HZIO(306, "HZIO"),
+    WPIO(307, "WPIO"),
+    LIN_ANSAUG_VERDAMPFER(308, "LIN_ANSAUG_VERDAMPFER"),
+    LIN_MULTI1(309, "LIN_MULTI1"),
+    LIN_MULTI2(310, "LIN_MULTI2"),
+    EINST_LEISTUNG_ZWE(311, "Einst_Leistung_ZWE"),
+    SPRACHEN_ESPANOL(312, "Sprachen_ESPANOL"),
+    TEMP_BW_OBEN(313, "Temp_BW_oben"),
+    MAXIO(314, "MAXIO"),
+    OUT_ABTAUWUNSCH(315, "OUT_Abtauwunsch"),
+    SMARTGRID(316, "SmartGrid"),
+    DREHZAHLGEREGELT(317, "Drehzahlgeregelt"),
+    P155_INVERTER(318, "P155_Inverter"),
+    LEISTUNGSFREIGABE(319, "Leistungsfreigabe"),
+    EINST_VORL_AKT_KUEHL(320, "Einst_Vorl_akt_Kuehl"),
+    EINST_ABTAUEN_IM_WARMWASSER(321, "Einst_Abtauen_im_Warmwasser"),
+    WAERMEMENGE_ZWE(32, "Waermemenge_ZWE");
```

Comment:
language as above.

Reply:
The enum names are those used in the internal java applet of the heat pump. Don't think it makes much sense to translate them.

Reply:
The question int this case is: Is this always the case? Is the language switchable in the device or can other people get different names? If yes, the more internationally used language should win. If the channels are even on outside of DACH sold devices are named in german, i it is ok to name them in german also here.

Reply:
Guess I didn't make that clear. Those are the variable names used inside the java applet code. Those aren't visible to the customer at all. The java applet itself is served with translation files. But as far as I have seen for some languages not all words are translated (even not for English). Some are still German and some translations are built by concatenating various parts, which imho might not be correct for all languages. But it's hard to say. The code of the applet is a jar file containing compiled classes...

Reply:
Do i understand right? Those enum constants are the reverse engineered internal variable names? Then, i guess, the german strings would be okay. But please document them then using Javadoc-comments to describe them in English.

I'm not fully sure if this is enough. There we need to wait on a second review of a more experienced reviewer. But i'm pretty sure just german names won't be enough.

Reply:
For those visibility values I actually don't have any proper descriptions. Not even in german. I'm guessing their meaning based on the return values of my heat pump and the internal names. But will add a comment to describe that.